packages/@react-stately/table/src/useTableColumnResizeState.ts

```diff
@@ -0,0 +1,137 @@
+
+import {ColumnProps} from '@react-types/table';
+import {getContentWidth, getDynamicColumnWidths, getMaxWidth, getMinWidth, isStatic, mapToArray, parseStaticWidth} from './utils';
+import {GridNode} from '@react-types/grid';
+import {Key, MutableRefObject, useCallback, useRef, useState} from 'react';
+import {useLayoutEffect} from '@react-aria/utils';
+
+type useTableColumnResizeType<T> = [
+  columnWidths: MutableRefObject<Map<Key, number>>,
+  resizeColumns: (column: GridNode<T>, newWidth: number, doneResizing?: boolean) => { key: Key, width: number }[],
+  setTableWidth: (width: number) => void,
+  onColumnResizeStart: () => void,
+  onColumnResizeEnd: () => void
+]
```

Comment:
So I think there is a problem with doing it this way. This type is actually for the return value of `useTableColumnResizeState`, which is following the tuple-styled return value of `[something, setSomething] = useState()`. So we wouldn't want to change it to an object because we're not trying to destructure from an object. 

These used to be inline but they were really vague as a result. So to make it more clear what the types are of what we're trying to return, I added this type and named each tuple value - even though the names don't actually do or mean anything. It's just there to help understand the type and nothing more.

Thoughts?

Reply:
I think that'd make more sense if there was only one argument to useTableColumnResizeState

`let [something, setSomething] = useState(defaultSomething)`

but this hook is taking two pieces of state, and returns 1 output state, 2? setters, and 2 onEvents
so it doesn't much feel like useState even though it says state in its name

it feels much more like how `useTableState` works, so I think I'd prefer it to take a single object for the argument with the named keys `columns` and `getDefaultWidth`
and returns an object with the named properities on it as you've already named them

Reply:
out of curiosity, why put onColumnResize/End both inside and outside of the useTableColumnResizeState, maybe it's better to do something like this

```
let {
  columnWidths,
  setTableWidth,
  onColumnResizeStart,
  onColumnResize,
  onColumnResizeEnd
} = useTableColumnResizeState({
  columns: collection.columns,
  getDefaultWidth,
  onColumnResizeEnd,
  onColumnResize
});
```

which could be simplified in the calling to

```
let columnResizingState = useTableColumnResizeState({
  columns: collection.columns,
  ...props
});
return {
  collection,
  disabledKeys,
  // ...
  ...columnResizingState
}
```

Reply:
That's fair. I'll look into it.

Reply:
> out of curiosity, why put onColumnResize/End both inside and outside of the useTableColumnResizeState

We do need to return these from the `useTableState` so that way `useTableColumnResize` can utilize it.

Reply:
Talked in person and resolved this